Bottom: fab347d02109b60b168ff984a4d2d21706413a80
Top:    e3a6b0dc76b8b55127c0f08826d1639b71e06ab4
Author: Ben Greear <greearb@candelatech.com>
Date:   2014-11-21 13:59:45 -0500

mac80211: print kernel logs for ibss join failures.

Helps user know what is wrong with config and/or driver.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/mac80211/ibss.c b/net/mac80211/ibss.c
index a31d307..f758736 100644
--- a/net/mac80211/ibss.c
+++ b/net/mac80211/ibss.c
@@ -1757,12 +1757,16 @@ int ieee80211_ibss_join(struct ieee80211_sub_if_data *sdata,
 	ret = cfg80211_chandef_dfs_required(local->hw.wiphy,
 					    &params->chandef,
 					    sdata->wdev.iftype);
-	if (ret < 0)
+	if (ret < 0) {
+		sdata_info(sdata, "ibss-join: chandef-dfs-required failed.\n");
 		return ret;
+	}
 
 	if (ret > 0) {
-		if (!params->userspace_handles_dfs)
+		if (!params->userspace_handles_dfs) {
+			sdata_info(sdata, "ibss-join: No userspace-handles-dfs set.\n");
 			return -EINVAL;
+		}
 		radar_detect_width = BIT(params->chandef.width);
 	}
 
@@ -1773,8 +1777,10 @@ int ieee80211_ibss_join(struct ieee80211_sub_if_data *sdata,
 	ret = ieee80211_check_combinations(sdata, &params->chandef, chanmode,
 					   radar_detect_width);
 	mutex_unlock(&local->chanctx_mtx);
-	if (ret < 0)
+	if (ret < 0) {
+		sdata_info(sdata, "ibss-join:  Failed iface combination check.\n");
 		return ret;
+	}
 
 	if (params->bssid) {
 		memcpy(sdata->u.ibss.bssid, params->bssid, ETH_ALEN);
